docs/security.rst

```diff
@@ -96,45 +155,70 @@ Multiple OAuth2 security schemes in AND fashion are not supported.
 
 .. _OpenAPI specification: https://swagger.io/docs/specification/authentication/#multiple
 
-Deploying Authentication
+Custom security handlers
 ------------------------
 
-Some production hosting environments, such as Apache with modwsgi, do not by default pass
-authentication headers to WSGI applications.  Therefore, to allow connexion to handle
-authentication, you will need to enable passthrough.
+You can implement your own security handlers for schemes that are not supported yet in Connexion
+by subclassing the ``connexion.security.AbstractSecurityHandler`` class and passing it in a custom
+``security_map`` to your application or API:
 
-Instructions for `enabling authentication passthrough in modwsgi`_ are available as
-part of the `modwsgi documentation`_.
+.. code-block:: python
+    :caption: **app.py**
 
-HTTPS Support
--------------
+    from connexion.security import AbstractSecurityHandler
 
-When specifying HTTPS as the scheme in the API YAML file, all the URIs
-in the served Swagger UI are HTTPS endpoints. The problem: The default
-server that runs is a "normal" HTTP server. This means that the
-Swagger UI cannot be used to play with the API. What is the correct
-way to start a HTTPS server when using Connexion?
 
-One way, `described by Flask`_, looks like this:
+    class MyCustomSecurityHandler(AbstractSecurityHandler):
 
-.. code-block:: python
+        security_definition_key = "x-{type}InfoFunc"
+        environ_key = "{TYPE}INFO_FUNC"
 
-   from OpenSSL import SSL
-   context = SSL.Context(SSL.SSLv23_METHOD)
-   context.use_privatekey_file('yourserver.key')
-   context.use_certificate_file('yourserver.crt')
+        def _get_verify_func(self, {type}_info_func):
+        ...
 
-   app.run(host='127.0.0.1', port='12344',
-           debug=False/True, ssl_context=context)
+    security_map = {
+        "{type}": MyCustomSecurityHandler,
+    }
 
-However, Connexion doesn't provide an ssl_context parameter. This is
-because Flask doesn't, either--but it uses ``**kwargs`` to send the
-parameters to the underlying `werkzeug`_ server.
+.. tab-set::
 
-.. _rfc6750: https://tools.ietf.org/html/rfc6750
-.. _rfc6749: https://tools.ietf.org/html/rfc6749
-.. _rfc7662: https://tools.ietf.org/html/rfc7662
-.. _minimal API Key example application: https://github.com/spec-first/connexion/tree/main/examples/apikey
-.. _minimal JWT example application: https://github.com/spec-first/connexion/tree/main/examples/jwt
-.. _enabling authentication passthrough in modwsgi: https://modwsgi.readthedocs.io/en/develop/configuration-directives/WSGIPassAuthorization.html
-.. _modwsgi documentation: https://modwsgi.readthedocs.io/en/develop/index.html
+    .. tab-item:: AsyncApp
+        :sync: AsyncApp
+
+        .. code-block:: python
+            :caption: **app.py**
+
+            from connexion import AsyncApp
+
+            app = AsyncApp(__name__, security_map=security_map)
+            app.add_api("openapi.yaml", security_map=security_map)
```

Comment:
Minor comment: clarify only one is needed?

Reply:
Was thinking about this as well, but only added it in the text for now.

I use this pattern all over the docs though, so if we change it, we should do it everywhere, and I would do it in a separate PR.

Reply:
Makes sense, better to keep it consistent indeed :)

Let's keep it for now and we can always clarify in the future if needed